### DISCUSSION TITLE
CompressedPublicKey vs bitcoin::PublicKey vs secp256k1::PublicKey?

### DISCUSSION THREAD

**Discussion by Kodylow** [fedimint/fedimint#1444] [General]
Any particular reason why we're using CompressedPublicKey in the wallet and some of the crypto or is that just legacy code that I can update to using PublicKey? Looks like the wallet hasn't been touched in a while, CompressedPublicKey might just be a hold over from an old release of rust bitcoin?

Within the bitcoin crate PublicKey as a struct looks like this:
```
pub struct PublicKey {
    /// Whether this public key should be serialized as compressed
    pub compressed: bool,
    /// The actual ECDSA key
    pub inner: secp256k1::PublicKey,
}
```
Within our wallet CompressedPublicKey looks like this:
```
pub struct CompressedPublicKey {
    pub key: secp256k1::PublicKey,
}
```

In my PR I already updated us to using secp256k1::PublicKey everywhere we can, can I just get rid of CompressedPublicKey and switch it to PublicKey directly? We're distinguishing already between XOnly vs Compressed because XOnlyPublicKeys are called XOnly

**Comment by justinmoon** [comment]
Certainly looks like we could just scrap our `CompressedPublicKey` struct. But I'll defer to @elsirion.

The only thing it seems to do is set `compressed: true` in `to_public_key()`

But I think `secp256k1::PublicKey::to_public_key()` [already does this](https://github.com/rust-bitcoin/rust-miniscript/blob/master/src/lib.rs#L301) because it calls `bitcoin::PublicKey::new()` which sets `compressed: true`.

**Comment by elsirion** [comment]
The reason it exists is that we
1. **NEVER** want to deal with uncompressed keys, that's legacy and should fail hard instead of being tolerated (maybe using the rust-bitcoin type and asserts would be more appropriate? this way we can't have any slip ups, so I chose it)
2. We can't implement `MiniscriptKey` for `secp256k1::PublicKey` and [the PR that did it in rust-miniscript](https://github.com/rust-bitcoin/rust-miniscript/commit/0888fe9ee625c9b7e8a8aff8d9bf4aa75f88b248) was only merged 4 months ago (after I wrote that workaround) and I wasn't aware of it till today.

EDIT: so great point @Kodylow, let's replace it with `secp256k1::PublicKey`.
